File: Data-Structure/_All-In-One-LeetCode/00001_twoSum (E)/twoSum.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>(); // we need to use this Map<Integer,Integer>
        for (int i = 0; i < nums.length; i++) {
            int key = target - nums[i];
            if (map.containsKey(key)) {
                return new int[] { map.get(key), i };
            } else {
                map.put(nums[i], i);
            }
        }
        return new int[] {};

    }
}

File: Data-Structure/_All-In-One-LeetCode/00001_twoSum (E)/twoSum.go
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	N := len(nums)

	for i := 0; i < N; i++ {
		key := target - nums[i]
		if _, val := m[key]; val {
			return []int{m[key], i}
		}
		m[nums[i]] = i
	}
	return nil
}